### PR TITLE
Added currently implemented API to the docs 4529

### PR DESCRIPTION
**Pull request by kalabiyau**

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/40776/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged 5556ed984bcb682cff8f313c81135fb68bf88d46 into 2e3dbe2feb2c18580c919c3f0d548c9ecb9e7d60 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/40777/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged 87c9ffa0ae78f8b5bd2f4f4e77ae0d972501d425 into 2e3dbe2feb2c18580c919c3f0d548c9ecb9e7d60 )

**Comment by suse-tests-pass** [comment]
@tmuntan1 will review your pull request :dancers: check https://trello.com/c/LTPiwYTQ/4529-move-documentation-for-connect-api-into-repository
  @tmuntan1: Please review this pull request using our guidelines:
  * test for acceptance criteria / functionality
  * check if the new code is covered with tests
  * check for unintended consequences
  * encourage usage of the boyscout rule
  * make sure the code is architected in the best way
  * check that no unnecessary technical debt got introduced
  * make sure that no unnecessary FIXMEs or TODOs got added

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/41563/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged 08fd5c006e4d7f4a56b7ca059fa3ccb6f72ff8a6 into 2e3dbe2feb2c18580c919c3f0d548c9ecb9e7d60 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/41893/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged b13ca015fd716d5c3f3ce8bb8873b0c3468af243 into 2e3dbe2feb2c18580c919c3f0d548c9ecb9e7d60 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/41894/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged d8439cd2a1b543c75d47415eb623c1aa182ca8b8 into 2e3dbe2feb2c18580c919c3f0d548c9ecb9e7d60 )

**Comment by kalabiyau** [comment]
@tmuntan1 thanks again for spotting these. Please check again.

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/41895/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged f3b529958f90e74ed7b9339e9b41a600bf3a3ea5 into 2e3dbe2feb2c18580c919c3f0d548c9ecb9e7d60 )

File: doc/SCC-API-RFC-(Draft).md [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+    - [list services](#sub_service_list)
+    - [list repositories](#sub_repo_list)
+  - [Systems](#systems)
+    - [show system](#sys_show)
+    - [list products](#sys_prod_list)
+    - [list repositories](#sys_repo_list)
+  - [Public](#public)
+    - [list products](#prod_list)
+    - [list services](#service_list)
+    - [list repositories](#repo_list)
```
Comment (tmuntan1) [inline doc/SCC-API-RFC-(Draft).md]: Anchor goes nowhere

File: doc/SCC-API-RFC-(Draft).md [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+### <a id="public">Public</a>
+For all endpoints on this level there is no authentication needed. They are
+publicly available. Of course, that means that only publicly available resources
+will be listed.
+
+The typical use case for these calls is the SLES installer
+
+  - [list products](#prod_list)
+  - [list services](#service_list)
+  - [list repositories](#repo_list)
```
Comment (tmuntan1) [inline doc/SCC-API-RFC-(Draft).md]: Anchor goes nowhere

File: doc/SCC-API-V3.md [hunk 3]
```diff
@@ -0,0 +1,920 @@
+**This is the documentation for version 3 of the API. Please check the docs of the [latest version](SCC-API-(Implemented))!**
```
Comment (tmuntan1) [inline doc/SCC-API-V3.md]: Link broken

File: doc/SCC-API-V3.md [hunk 4]
```diff
@@ -0,0 +1,920 @@
...
+
+## API versioning:
+By default, the latest available api version is used. Clients are recommended to 
+hardcode a fixed version in order not to break when the SCC api changes in an incompatible way. 
+This is done by setting the accept header in the request, for example: 
+
+```Accept: application/vnd.scc.suse.com.v1+json```
+
+The current API version is 3. Documentation of older api versions: 
+* [Version 1](SCC-API-V1)
```
Comment (tmuntan1) [inline doc/SCC-API-V3.md]: Link broken

File: doc/SCC-API-V3.md [hunk 5]
```diff
@@ -0,0 +1,920 @@
...
+* [Version 2](SCC-API-V2)
```
Comment (tmuntan1) [inline doc/SCC-API-V3.md]: Link broken

File: doc/YaST-abstraction-layer.md [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+- [update system](#update-system)
+- [activate product](#activate-product)
+- [upgrade product](#upgrade-product)
+- [downgrade product](#downgrade-product)
+- [synchronize](#synchronize)
+- [show product](#show-product)
+- [check product activation](#product-activated?)
+- [system migrations](#system-migrations)
+- [activated_products](#activated-products)
+- [add service](#add-service)
```
Comment (tmuntan1) [inline doc/YaST-abstraction-layer.md]: anchor broken

File: doc/YaST-abstraction-layer.md [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+- [remove service](#remove-service)
```
Comment (tmuntan1) [inline doc/YaST-abstraction-layer.md]: anchor broken

File: doc/SCC-API-RFC-(Draft).md (hunk 2 above)
Comment (kalabiyau) [inline doc/SCC-API-RFC-(Draft).md]: Draft we can ignore as that was just a starting point back then. We will keep it for the history, but there is no point in fixing it from my POV.

**Comment by kalabiyau** [comment]
@tmuntan1 fixed, please check again

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/42128/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged d5b1d4b968d0b83d85b9af3e799489a7b4c4db1a into 2e3dbe2feb2c18580c919c3f0d548c9ecb9e7d60 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/42129/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged 1642459e0a7cd9c0ded80c87966521c4bbc9d204 into 2e3dbe2feb2c18580c919c3f0d548c9ecb9e7d60 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/45873/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged bc8b7e1cb9e75534d7e50e0fc190c9ff00c696ef into 2e3dbe2feb2c18580c919c3f0d548c9ecb9e7d60 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/45905/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged 8e76c63e3b2e6a4da2108c6cf79c0f5fbfe99fd6 into 2e3dbe2feb2c18580c919c3f0d548c9ecb9e7d60 )

**Pull request merged**